src/components/Timer/Timer.js

```diff
@@ -0,0 +1,32 @@
+import './Timer.css';
+import TextTimer from '../TextTimer/TextTimer';
+import LightsaberTimer from '../LightsaberTimer/LightsaberTimer';
+
+export default function Timer(callback = () => null) {
+    let minute = 1;
+    let sec = 30;
+    let allSeconds = (minute * 60 + sec);
+    const timerArea = document.createElement('div');
+    timerArea.className = 'timerArea';
+
+    let interval = window.setInterval(function () {
+        sec--;
+
+        if (minute != 0 && sec === 0) {
+            minute--;
+            sec = 59;
+        }
+        if (minute === 0 && sec === 0) {
+            clearInterval(interval);
+            callback();
+        }
+
+        let remainingSeconds = Math.round(((minute * 60 + sec) / allSeconds * 100), 2);
```

Comment:
I am not sure why you calculate here `minute` and `sec` and divide it by `allSeconds` which points to `minute` and `sec`

Reply:
Sorry, I assigned wrong name variable.  It is a progress bar value in percent. I renamed variable name to "progressValue".

Reply:
np ;d